diff --git hack/local-up-cluster.sh hack/local-up-cluster.sh
index f1cba1f..4d569ba 100755
--- hack/local-up-cluster.sh
+++ hack/local-up-cluster.sh
@@ -284,6 +284,8 @@ function start_kubelet {
       fi
 
       sudo -E "${GO_OUT}/kubelet" ${priv_arg}\
+        --network-plugin=cni \
+        --network-plugin-dir="/etc/cni/net.d" \
         --v=${LOG_LEVEL} \
         --chaos-chance="${CHAOS_CHANCE}" \
         --container-runtime="${CONTAINER_RUNTIME}" \