diff --git a/contrib/vagrant/start.sh b/contrib/vagrant/start.sh
index 15def2284..f6a4f43f6 100755
--- a/contrib/vagrant/start.sh
+++ b/contrib/vagrant/start.sh
@@ -42,7 +42,7 @@ export 'VM_BASENAME'=${K8S+"k8s"}
 # Sets the default cilium TUNNEL_MODE to "vxlan"
-export 'TUNNEL_MODE_STRING'=${TUNNEL_MODE_STRING:-"-t vxlan"}
+export 'TUNNEL_MODE_STRING'=${TUNNEL_MODE_STRING:-"-t disabled"}
 # Replies Yes to all prompts asked in this script
 export 'YES_TO_ALL'=${YES_TO_ALL:-"0"}
 
@@ -203,8 +203,8 @@ function write_k8s_install() {
         k8s_cluster_api_server_ip="FD03::1"
         k8s_cluster_dns_ip="FD03::A"
     fi
-    k8s_cluster_cidr=${k8s_cluster_cidr:-"10.16.0.0/12"}
-    k8s_node_cidr_mask_size=${k8s_node_cidr_mask_size:-"16"}
+    k8s_cluster_cidr=${k8s_cluster_cidr:-"10.244.0.0/16"}
+    k8s_node_cidr_mask_size=${k8s_node_cidr_mask_size:-"24"}
     k8s_service_cluster_ip_range=${k8s_service_cluster_ip_range:-"172.20.0.0/24"}
     k8s_cluster_api_server_ip=${k8s_cluster_api_server_ip:-"172.20.0.1"}
     k8s_cluster_dns_ip=${k8s_cluster_dns_ip:-"172.20.0.10"}
@@ -278,7 +278,7 @@ function write_cilium_cfg() {
     ipv6_addr="${3}"
     filename="${4}"
 
-    cilium_options="--auto-direct-node-routes"
+    cilium_options="--flannel-master-device=cni0 --auto-direct-node-routes"
 
     if [[ "${IPV4}" -eq "1" ]]; then
         if [[ -z "${K8S}" ]]; then
diff --git a/examples/kubernetes-ingress/scripts/helpers.bash b/examples/kubernetes-ingress/scripts/helpers.bash
index bdec39577..5cbcc2d87 100644
--- a/examples/kubernetes-ingress/scripts/helpers.bash
+++ b/examples/kubernetes-ingress/scripts/helpers.bash
@@ -55,9 +55,9 @@ kubernetes_master="${controllers_ips[0]}"
 # Default values for IPv4
 #
 # CIDR Range for Pods in cluster.
-k8s_cluster_cidr=${K8S_CLUSTER_CIDR:-"10.16.0.0/12"} # 10.16.0.1-10.31.255.255
+k8s_cluster_cidr=${K8S_CLUSTER_CIDR:-"10.244.0.0/16"} # 10.16.0.1-10.31.255.255
 # Mask size for node cidr in cluster.
-k8s_node_cidr_mask_size=${K8S_NODE_CIDR_MASK_SIZE:-"16"} # 1st Node: 10.16.0.1-10.16.255.254, 2nd Node: 10.17.0.1-10.17.255.254...
+k8s_node_cidr_mask_size=${K8S_NODE_CIDR_MASK_SIZE:-"24"} # 1st Node: 10.16.0.1-10.16.255.254, 2nd Node: 10.17.0.1-10.17.255.254...
 # CIDR Range for Services in cluster.
 k8s_service_cluster_ip_range=${K8S_SERVICE_CLUSTER_IP_RANGE:-"172.20.0.0/24"}
 cluster_dns_ip=${K8S_CLUSTER_DNS_IP:-"172.20.0.10"}
diff --git a/plugins/cilium-cni/Makefile b/plugins/cilium-cni/Makefile
index 166a8fa98..a6ff6059b 100644
--- a/plugins/cilium-cni/Makefile
+++ b/plugins/cilium-cni/Makefile
@@ -18,6 +18,6 @@ $(TARGET): $(SOURCES)
 
 install:
 	$(QUIET)$(INSTALL) -m 0755 -d $(DESTDIR)$(CNICONFDIR)
-	$(QUIET)$(INSTALL) -m 0644 05-cilium-cni.conf $(DESTDIR)$(CNICONFDIR)
+	$(QUIET)$(INSTALL) -m 0644 04-flannel-cilium-cni.conflist $(DESTDIR)$(CNICONFDIR)
 	$(QUIET)$(INSTALL) -m 0755 -d $(DESTDIR)$(CNIBINDIR)
 	$(QUIET)$(INSTALL) -m 0755 $(TARGET) $(DESTDIR)$(CNIBINDIR)
diff --git a/test/provision/wait-cilium.sh b/test/provision/wait-cilium.sh
index e9d396a3c..07136746f 100755
--- a/test/provision/wait-cilium.sh
+++ b/test/provision/wait-cilium.sh
@@ -2,16 +2,10 @@
 
 main() {
     local cilium_started
-    cilium_started=false
 
-    for ((i = 0 ; i < 24; i++)); do
-        if cilium status --brief > /dev/null 2>&1; then
-            cilium_started=true
-            break
-        fi
-        sleep 5s
-        echo "Waiting for Cilium daemon to come up..."
-    done
+    cilium_started=true
+    echo "As Cilium is running in visibility-mode we can't wait for it to start"
+    echo "as it might be waiting for cni0 to be created."
 
     if [ "$cilium_started" = true ] ; then
         echo 'Cilium successfully started!'
