diff --git a/vendor/k8s.io/code-generator/cmd/informer-gen/generators/packages.go b/vendor/k8s.io/code-generator/cmd/informer-gen/generators/packages.go
index 049f82476..e936e29f0 100644
--- a/vendor/k8s.io/code-generator/cmd/informer-gen/generators/packages.go
+++ b/vendor/k8s.io/code-generator/cmd/informer-gen/generators/packages.go
@@ -283,9 +283,10 @@ func factoryInterfacePackage(basePackage string, boilerplate []byte, clientSetPa
 
 func groupPackage(basePackage string, groupVersions clientgentypes.GroupVersions, boilerplate []byte) generator.Package {
 	packagePath := filepath.Join(basePackage, groupVersions.PackageName)
+	groupPkgName := strings.Split(string(groupVersions.PackageName), ".")[0]
 
 	return &generator.DefaultPackage{
-		PackageName: groupVersions.PackageName,
+		PackageName: groupPkgName,
 		PackagePath: packagePath,
 		HeaderText:  boilerplate,
 		GeneratorFunc: func(c *generator.Context) (generators []generator.Generator) {
diff --git a/vendor/k8s.io/code-generator/hack/update-codegen.sh b/vendor/k8s.io/code-generator/hack/update-codegen.sh
index 11ed17ad6..3ba3f8e58 100755
--- a/vendor/k8s.io/code-generator/hack/update-codegen.sh
+++ b/vendor/k8s.io/code-generator/hack/update-codegen.sh
@@ -26,7 +26,7 @@ SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 #   instead of the $GOPATH directly. For normal projects this can be dropped.
 "$(dirname "${BASH_SOURCE[0]}")"/../generate-internal-groups.sh all \
   k8s.io/code-generator/_examples/apiserver k8s.io/code-generator/_examples/apiserver/apis k8s.io/code-generator/_examples/apiserver/apis \
-  "example:v1 example2:v1" \
+  "example:v1 example2.io:v1" \
   --output-base "$(dirname "${BASH_SOURCE[0]}")/../../.." \
   --go-header-file "${SCRIPT_ROOT}/hack/boilerplate.go.txt"
 "$(dirname "${BASH_SOURCE[0]}")"/../generate-groups.sh all \
