diff --git a/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.in b/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.in
index bba39a1..db5bf52 100644
--- a/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.in
+++ b/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.in
@@ -27,6 +27,7 @@ metadata:
     k8s-app: kube-dns
     version: v17.1
     kubernetes.io/cluster-service: "true"
+    io.cilium.admin.kube-system: ""
 spec:
   replicas: {{ pillar['dns_replicas'] }}
   selector:
@@ -38,6 +39,7 @@ spec:
         k8s-app: kube-dns
         version: v17.1
         kubernetes.io/cluster-service: "true"
+        io.cilium.admin.kube-system: ""
     spec:
       containers:
       - name: kubedns
diff --git a/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.sed b/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.sed
index 15b26ab..4bb5f7a 100644
--- a/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.sed
+++ b/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.sed
@@ -27,6 +27,7 @@ metadata:
     k8s-app: kube-dns
     version: v17.1
     kubernetes.io/cluster-service: "true"
+    io.cilium.admin.kube-system: ""
 spec:
   replicas: $DNS_REPLICAS
   selector:
@@ -38,6 +39,7 @@ spec:
         k8s-app: kube-dns
         version: v17.1
         kubernetes.io/cluster-service: "true"
+        io.cilium.admin.kube-system: ""
     spec:
       containers:
       - name: kubedns
diff --git a/cluster/saltbase/salt/kube-dns/skydns-svc.yaml.in b/cluster/saltbase/salt/kube-dns/skydns-svc.yaml.in
index f8d6444..7aaa57c 100644
--- a/cluster/saltbase/salt/kube-dns/skydns-svc.yaml.in
+++ b/cluster/saltbase/salt/kube-dns/skydns-svc.yaml.in
@@ -27,6 +27,7 @@ metadata:
     k8s-app: kube-dns
     kubernetes.io/cluster-service: "true"
     kubernetes.io/name: "KubeDNS"
+    io.cilium.admin.kube-system: ""
 spec:
   selector:
     k8s-app: kube-dns
diff --git a/cluster/saltbase/salt/kube-dns/skydns-svc.yaml.sed b/cluster/saltbase/salt/kube-dns/skydns-svc.yaml.sed
index 837d72b..b94b2e9 100644
--- a/cluster/saltbase/salt/kube-dns/skydns-svc.yaml.sed
+++ b/cluster/saltbase/salt/kube-dns/skydns-svc.yaml.sed
@@ -27,6 +27,7 @@ metadata:
     k8s-app: kube-dns
     kubernetes.io/cluster-service: "true"
     kubernetes.io/name: "KubeDNS"
+    io.cilium.admin.kube-system: ""
 spec:
   selector:
     k8s-app: kube-dns
diff --git a/hack/local-up-cluster.sh b/hack/local-up-cluster.sh
index 6a7e1e6..1328e6e 100755
--- a/hack/local-up-cluster.sh
+++ b/hack/local-up-cluster.sh
@@ -27,6 +27,7 @@ RUNTIME_CONFIG=${RUNTIME_CONFIG:-""}
 NET_PLUGIN=${NET_PLUGIN:-""}
 NET_PLUGIN_DIR=${NET_PLUGIN_DIR:-""}
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+SERVICE_CLUSTER_IP_RANGE=${SERVICE_CLUSTER_IP_RANGE:-"10.0.0.0/24"}
 # We disable cluster DNS by default because this script uses docker0 (or whatever
 # container bridge docker is currently using) and we don't know the IP of the
 # DNS pod to pass in as --cluster-dns. To set this up by hand, set this flag
@@ -283,7 +284,7 @@ function start_apiserver {
       --insecure-port="${API_PORT}" \
       --advertise-address="${API_HOST_IP}" \
       --etcd-servers="http://${ETCD_HOST}:${ETCD_PORT}" \
-      --service-cluster-ip-range="10.0.0.0/24" \
+      --service-cluster-ip-range="${SERVICE_CLUSTER_IP_RANGE}" \
       --cloud-provider="${CLOUD_PROVIDER}" \
       --cors-allowed-origins="${API_CORS_ALLOWED_ORIGINS}" >"${APISERVER_LOG}" 2>&1 &
     APISERVER_PID=$!
@@ -394,11 +395,11 @@ function start_kubelet {
 
 function start_kubeproxy {
     PROXY_LOG=/tmp/kube-proxy.log
-    sudo -E "${GO_OUT}/hyperkube" proxy \
-      --v=${LOG_LEVEL} \
-      --hostname-override="${HOSTNAME_OVERRIDE}" \
-      --master="http://${API_HOST}:${API_PORT}" >"${PROXY_LOG}" 2>&1 &
-    PROXY_PID=$!
+    #sudo -E "${GO_OUT}/hyperkube" proxy \
+    #  --v=${LOG_LEVEL} \
+    #  --hostname-override="${HOSTNAME_OVERRIDE}" \
+    #  --master="http://${API_HOST}:${API_PORT}" >"${PROXY_LOG}" 2>&1 &
+    #PROXY_PID=$!
 
     SCHEDULER_LOG=/tmp/kube-scheduler.log
     sudo -E "${GO_OUT}/hyperkube" scheduler \
