diff --git a/hack/lib/etcd.sh b/hack/lib/etcd.sh
index 11cd2b5..d39f9ac 100644
--- a/hack/lib/etcd.sh
+++ b/hack/lib/etcd.sh
@@ -42,8 +42,8 @@ kube::etcd::start() {

   # Start etcd
   ETCD_DIR=$(mktemp -d 2>/dev/null || mktemp -d -t test-etcd.XXXXXX)
-  kube::log::info "etcd -data-dir ${ETCD_DIR} --bind-addr ${host}:${port} >/dev/null 2>/dev/null"
-  etcd -data-dir ${ETCD_DIR} --bind-addr ${host}:${port} >/dev/null 2>/dev/null &
+  kube::log::info "etcd -addr ${host}:${port} -data-dir ${ETCD_DIR} --bind-addr ${host}:${port} >/dev/null 2>/dev/null"
+  etcd -addr ${host}:${port} -data-dir ${ETCD_DIR} --bind-addr ${host}:${port} >/tmp/etcdstdout 2>/tmp/etcdstderr &
   ETCD_PID=$!

   echo "Waiting for etcd to come up."
diff --git a/hack/lib/util.sh b/hack/lib/util.sh
index b42f4fa..67d992a 100755
--- a/hack/lib/util.sh
+++ b/hack/lib/util.sh
@@ -38,7 +38,7 @@ kube::util::wait_for_url() {
   local i
   for i in $(seq 1 $times); do
     local out
-    if out=$(curl -fs $url 2>/dev/null); then
+    if out=$(curl -gfs $url 2>/dev/null); then
       kube::log::status "On try ${i}, ${prefix}: ${out}"
       return 0
     fi
diff --git a/hack/local-up-cluster.sh b/hack/local-up-cluster.sh
index 7173518..ab005dc 100755
--- a/hack/local-up-cluster.sh
+++ b/hack/local-up-cluster.sh
@@ -260,13 +260,14 @@ function start_apiserver {
     APISERVER_LOG=/tmp/kube-apiserver.log
     sudo -E "${GO_OUT}/kube-apiserver" ${priv_arg} ${runtime_config}\
       --v=${LOG_LEVEL} \
+      --advertise-address="${NODE_IP}" \
       --cert-dir="${CERT_DIR}" \
       --service-account-key-file="${SERVICE_ACCOUNT_KEY}" \
       --service-account-lookup="${SERVICE_ACCOUNT_LOOKUP}" \
       --admission-control="${ADMISSION_CONTROL}" \
       --insecure-bind-address="${API_HOST}" \
       --insecure-port="${API_PORT}" \
-      --etcd-servers="http://127.0.0.1:4001" \
+      --etcd-servers="http://${ETCD_HOST}:4001" \
       --service-cluster-ip-range="10.0.0.0/24" \
       --cors-allowed-origins="${API_CORS_ALLOWED_ORIGINS}" >"${APISERVER_LOG}" 2>&1 &
     APISERVER_PID=$!
@@ -311,7 +312,7 @@ function start_kubelet {
 	 fi
       fi
       # Enable dns
-      if [[ "${ENABLE_CLUSTER_DNS}" = true ]]; then
+      if [[ "${ENABLE_DNS}" = true ]]; then
          dns_args="--cluster-dns=${DNS_SERVER_IP} --cluster-domain=${DNS_DOMAIN}"
       else
          # To start a private DNS server set ENABLE_CLUSTER_DNS and
@@ -337,7 +338,7 @@ function start_kubelet {
         --rkt-path="${RKT_PATH}" \
         --rkt-stage1-image="${RKT_STAGE1_IMAGE}" \
         --hostname-override="${HOSTNAME_OVERRIDE}" \
-        --address="127.0.0.1" \
+        --address="${NODE_IP}" \
         --api-servers="${API_HOST}:${API_PORT}" \
         --cpu-cfs-quota=${CPU_CFS_QUOTA} \
         ${dns_args} \
